tests/unit/test_args.py

```diff
@@ -130,3 +130,48 @@ def test_max_pool_from_client_config_forwarded_to_endpoint_creator(self):
                 response_parser_factory=None, timeout=(60, 60), verify=True,
                 max_pool_connections=20
             )
+
+    def test_s3_with_endpoint_url_still_resolves_region(self):
+        self.args_create = args.ClientArgsCreator(
+            mock.Mock(), None, None, None, None)
+        service_model = mock.Mock()
+        service_model.endpoint_prefix = 's3'
+        service_model.metadata = {'protocol': 'rest-xml'}
+        config = botocore.config.Config()
+        bridge = mock.Mock()
+        bridge.resolve.side_effect = [
+            {
+                'region_name': None, 'signature_version': 's3v4',
+                'endpoint_url': 'http://other.com/', 'signing_name': 's3',
+                'signing_region': None, 'metadata': {}
+            },
+            {
+                'region_name': 'us-west-2', 'signature_version': 's3v4',
+                'enpoint_url': 'https://s3-us-west-2.amazonaws.com',
+                'signing_name': 's3', 'signing_region': 'us-west-2',
+                'metadata': {}
+            }
+        ]
+        client_args = self.args_create.get_client_args(
+            service_model, 'us-west-2', True, 'http://other.com/', True, None,
+            {}, config, bridge)
+        self.assertEqual(
+            client_args['client_config'].region_name, 'us-west-2')
+
+    def test_region_does_not_resolve_if_not_s3_and_endpoint_url_provided(self):
```

Comment:
It tests that no region is inferred when you provide an endpoint url for non-s3 services.